File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/dataset/create/LinkedDataCreation.java
/*
 * Copyright 2013 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.create;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonProperty;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.externaldms.id.IExternalDmsId;
import ch.systemsx.cisd.base.annotation.JsonObject;
import ch.systemsx.cisd.common.annotation.TechPreview;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.dataset.create.LinkedDataCreation")
public class LinkedDataCreation implements ICreation
{
    private static final long serialVersionUID = 1L;

    /**
     * @deprecated Use {@link ContentCopyCreation} instead.
     */
    @Deprecated
    @JsonProperty
    private String externalCode;

    /**
     * @deprecated Use {@link ContentCopyCreation} instead.
     */
    @Deprecated
    @JsonProperty
    private IExternalDmsId externalDmsId;

    private List<ContentCopyCreation> contentCopies;

    public String getExternalCode()
    {
        return externalCode;
    }

    public void setExternalCode(String externalCode)
    {
        this.externalCode = externalCode;
    }

    public IExternalDmsId getExternalDmsId()
    {
        return externalDmsId;
    }

    public void setExternalDmsId(IExternalDmsId externalDmsId)
    {
        this.externalDmsId = externalDmsId;
    }

    public List<ContentCopyCreation> getContentCopies()
    {
        return contentCopies;
    }

    public void setContentCopies(List<ContentCopyCreation> contentCopies)
    {
        this.contentCopies = contentCopies;
    }
}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/dataset/create/LinkedDataCreation.js
define([ "stjs" ], function(stjs) {
	var LinkedDataCreation = function() {
	};
	stjs.extend(LinkedDataCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.dataset.create.LinkedDataCreation';
		constructor.serialVersionUID = 1;
		prototype.externalCode = null;
		prototype.externalDmsId = null;
		prototype.contentCopies = null;

		prototype.getExternalCode = function() {
			return this.externalCode;
		};
		prototype.setExternalCode = function(externalCode) {
			this.externalCode = externalCode;
		};
		
		prototype.getExternalDmsId = function() {
			return this.externalDmsId;
		};
		prototype.setExternalDmsId = function(externalDmsId) {
			this.externalDmsId = externalDmsId;
		};
		
		prototype.getContentCopies = function() {
			return this.contentCopies;
		};
		prototype.setContentCopies = function(contentCopies) {
			this.contentCopies = contentCopies;
		};
	}, {
		externalDmsId : "IExternalDmsId",
		contentCopies : {
			name : "List",
			arguments : [ "Object" ]
		}
	});
	return LinkedDataCreation;
})		
